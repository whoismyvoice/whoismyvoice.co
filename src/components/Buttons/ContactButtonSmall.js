import React from 'react';
import PropTypes from 'prop-types';

// Styles
import '../../styles/ContactButtonSmall.css';

const ContactButtonSmall = ({
  add_style,
  icon,
  link,
  text,
}) => {
  return <a href={link} target="_blank">
    <button className={`contact__button_small ${add_style}`}>
      <div className={`button__small__icon ${icon}`} />
      <div className="button__small__text">
        {text}
      </div>
    </button>
  </a>;
};

ContactButtonSmall.propTypes = {
  add_style: PropTypes.string,
  icon: PropTypes.string,
  link: PropTypes.string,
  text: PropTypes.string,
};

export default ContactButtonSmall;
