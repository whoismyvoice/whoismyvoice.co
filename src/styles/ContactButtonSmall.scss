@import 'variables';
@import 'transitions';

.contact__button_small {
  @include transition(0.5s);
  width: auto;
  margin-left: 1%;
  margin-right: 1%;
  margin-bottom: 10px;
  height: 43px;
  display: inline-block;
  background-color: inherit;
  background-color: rgba(255, 255, 255, 0.1);
  border: 2px solid rgba(255, 255, 255, 0.3);
  border-radius: 4px;

  &:hover {
    background-color: rgba(255, 255, 255, 0.3);
    border: 2px solid rgba(255, 255, 255, 0.7);
  }

  @media screen and (max-width: $iphone6) {
    margin-left: 0;
    margin-right: 0;

  }

  &.smaller {
    width: 100%;
    min-width: 115px;
    max-width: 115px;

    @media screen and (max-width: $iphone6) {
      min-width: 110px;
      max-width: 115px;
    }
  }
}

.button__small__text {
  display: inline-block;
  color: $white;
  z-index: 2;
  margin-top: 6px;
  position: relative;
  font-weight: 600;
  font-size: 1.1em;

  @media screen and (max-width: $iphone6) {
    font-size: 1em;
  }
}

.button__small__icon {
  height: 40px;
  width: 40px;
  float: left;
  display: inline-block;

  @media screen and (max-width: $iphone6) {
    margin-left: 0;
    margin-right: 5px;
  }

  @media screen and (max-width: $iphone5-min) {
    margin-left: -5px;
  }

  &.twitter {
    background-image: url('../img/icons/twitter.png');
    background-position: center;
    background-size: contain;
  }

  &.phone {
    background-image: url('../img/icons/phone.png');
    background-position: center;
    background-size: contain;
  }

  &.email {
    background-image: url('../img/icons/email.png');
    background-position: center;
    background-size: contain;
  }

  &.vote {
    background-image: url('../img/icons/vote.png');
    background-position: center;
    background-size: contain;
  }
}

.button__text {
  display: block;
  width: 100%;
  margin-top: 7.5px;
  text-align: left;
  font-size: 16px;
  font-weight: 600;
  font-family: 'Open Sans';
  color: $white;
}
