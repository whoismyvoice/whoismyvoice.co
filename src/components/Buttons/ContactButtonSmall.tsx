import * as React from 'react';

// Styles
import '../../styles/ContactButtonSmall.css';

interface Props {
  addStyle?: string;
  icon: string;
  link: string;
  text: string;
}

const ContactButtonSmall = ({ addStyle, icon, link, text }: Props) => {
  return (
    <a href={link} target="_blank" rel="noopener noreferrer">
      <button className={`contact__button_small ${addStyle}`}>
        <div className={`button__small__icon ${icon}`} />
        <div className="button__small__text">{text}</div>
      </button>
    </a>
  );
};

export default ContactButtonSmall;
