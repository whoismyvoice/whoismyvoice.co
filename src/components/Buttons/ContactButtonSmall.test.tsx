import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import ContactButtonSmall from './ContactButtonSmall';

jest.mock('mixpanel-browser');

const props = {
  icon: 'icon',
  link: 'link',
  text: 'text',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ContactButtonSmall {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<ContactButtonSmall {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
