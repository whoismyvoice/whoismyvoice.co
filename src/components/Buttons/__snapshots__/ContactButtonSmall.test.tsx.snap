// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="link"
  rel="noopener noreferrer"
  target="_blank"
>
  <button
    className="contact__button_small undefined"
  >
    <div
      className="button__small__icon icon"
    />
    <div
      className="button__small__text"
    >
      text
    </div>
  </button>
</a>
`;
